core/sql/exp/ExpLOBaccess.cpp

```diff
@@ -1359,22 +1361,26 @@ Ex_Lob_Error ExLob::openDataCursor(char *file, LobsCursorType type, Int64 range,
     it = lobCursors_.find(string(file, strlen(file))); // to get the actual cursor object in the map
 
     if (!fdData_ || (openFlags_ != O_RDONLY)) 
-    {
-      hdfsCloseFile(fs_, fdData_);
-      fdData_ = NULL;
-      openFlags_ = O_RDONLY;
-      fdData_ = hdfsOpenFile(fs_, lobDataFile_, openFlags_, 0, 0, 0);
-      if (!fdData_) {
-        openFlags_ = -1;
-        lobCursorLock_.unlock();
-        return LOB_DATA_FILE_OPEN_ERROR;
+      {
+        hdfsCloseFile(fs_, fdData_);
+        fdData_ = NULL;
+        openFlags_ = O_RDONLY;
+        fdData_ = hdfsOpenFile(fs_, lobDataFile_, openFlags_, 0, 0, 0);
+       
+        if (!fdData_)
+          {
+            openFlags_ = -1;
+            *hdfsDetailError = errno;
```

Comment:
In the function prototype, hdfsDetailError defaults to NULL, So, should we check for null before attempting to store errno in there? (An alternative would be to make hdfsDetailError a reference parameter instead of a pointer; then the caller always has to provide it.)